Code/Framework/AzCore/Tests/Math/IntersectionTests.cpp

```diff
@@ -1297,4 +1297,96 @@ namespace UnitTest
 
         EXPECT_EQ(intersections, 0);
     }
-}
+
+    class MATH_ClipRayWithAabbTest : public AllocatorsFixture
+    {
+    };
+
+    TEST_F(MATH_ClipRayWithAabbTest, RayDoesNotGetClippedWhenMissingAabb)
+    {
+        // Test a ray that points straight down and completely misses a box.
+
+        const Aabb aabb = Aabb::CreateCenterRadius(AZ::Vector3::CreateZero(), 10.0f);
+
+        Vector3 rayStart(50.0f, 50.0f, 20.0f);
+        Vector3 rayEnd(50.0f, 50.0f, -20.0f);
+
+        float tClipStart = AZStd::numeric_limits<float>::max();
+        float tClipEnd = AZStd::numeric_limits<float>::max();
+
+        bool clipped = Intersect::ClipRayWithAabb(aabb, rayStart, rayEnd, tClipStart, tClipEnd);
+
+        // We expect the ray not to get clipped. There are no guarantees on the validity of any of the output values
+        // (rayStart, rayEnd, tClipStart, tClipEnd), so we won't validate them here.
+        EXPECT_FALSE(clipped);
+    }
+
+    TEST_F(MATH_ClipRayWithAabbTest, RayIsClippedWithAabb)
+    {
+        // Test a ray that starts 10 above a box, points straight down, and continues to 10 below the box.
+        // The z values should get clipped to the box.
+
+        const Aabb aabb = Aabb::CreateCenterRadius(AZ::Vector3::CreateZero(), 10.0f);
+
+        Vector3 rayStart(5.0f, 5.0f, 20.0f);
+        Vector3 rayEnd(5.0f, 5.0f, -20.0f);
+
+        float tClipStart = AZStd::numeric_limits<float>::max();
+        float tClipEnd = AZStd::numeric_limits<float>::max();
+
+        bool clipped = Intersect::ClipRayWithAabb(aabb, rayStart, rayEnd, tClipStart, tClipEnd);
+
+        // We expect the ray to get clipped.
+        EXPECT_TRUE(clipped);
+        EXPECT_EQ(rayStart, AZ::Vector3(5.0f, 5.0f, 10.0f));
+        EXPECT_EQ(rayEnd, AZ::Vector3(5.0f, 5.0f, -10.0f));
```

Comment:
update: It would likely be good to update this to `EXPECT_THAT(rayStart, IsClose(AZ::Vector3(...));` as that will do an epsilon check and also give more informative output if it fails

Reply:
5.0 and 10.0 can be represented exactly by floating point but it's possible error may be introduced in `ClipRayWithAabb` and so this test has a higher chance of failing with other values in future potentially